marlowe-playground-client/src/Examples/JS/Contracts.purs

```diff
@@ -199,3 +199,149 @@ makeDeposit(alice, 10,
            makePayment(bob, alice,
                Close))))
 """
+
+cfd :: String
+cfd =
+  """
+
+const party : Party = Role("party")
+
+const counterParty = Role("counterparty")
+
+const oracle: Party = Role("oracle")
+
+function waitForEvent (action : Action, timeout : number,  alternative : Contract): (_: Contract) => Contract {
+    return cont => When([Case(action, cont)], timeout, alternative)
+} 
+
+function before(time: number): number {
+    return time
+}    
+function orElse(contract: Contract): Contract {
+    return contract
+}
+
+function receiveValue(val: string): Action {
+    return Choice(ChoiceId(val, oracle), [Bound(0, 100)])
+}
+
+function readValue(val : string): Value {
+    return ChoiceValue(ChoiceId(val, oracle))
+}
+
+function waitFor(delay: number): (_: Contract) => Contract {
+    return cont => When([], delay, cont)
+}
+
+function checkIf(obs: Observation, c1: Contract, c2: Contract): Contract {
+    return If(obs, c1, c2)
+}
+
+function thenDo(c: Contract): Contract {
+    return c
+}
+
+function elseDo(c: Contract): Contract {
+    return c
+}
+
+
+function letValue(name: string, v: Value): (_: Contract) => Contract {
+    return cont => Let (ValueId(name), v, cont)
+}
+
+function useValue(name: string): Value {
+    return UseValue(name)
+}
+
+function value(v: number): Value {
+    return Constant(v)
+}
```

Comment:
It would be good to use bigint everywhere now that we have them. It is a matter of changing every number literal to have  an `n` at the end like `123n`, that way we set a better example.

Reply:
I am doing that for all other examples

Reply:
I'll do that!
I thought there would be an error if number is higher than MAX_SAFE_INT anyways

Reply:
Yes, but you could have a calculation that returns something between `MAX_SAFE_INT` and `-MAX_SAFE_INT`, but at some point uses something outside and thus the result is imprecise `(MAX_SAFE_INT^2)^(1/3)` or something like that.

Reply:
I guess there could be a practical case where we'd take a constant outside, like some complicated templating aka ACTUS.

In any case - I've corrected that.